pkg/volume/util/util.go

```diff
@@ -545,6 +545,6 @@ func GetPluginMountDir(host volume.VolumeHost, name string) string {
 // volume vs. some other type
 func IsLocalEphemeralVolume(volume v1.Volume) bool {
 	return volume.GitRepo != nil ||
-		(volume.EmptyDir != nil && volume.EmptyDir.Medium != v1.StorageMediumMemory) ||
+		(volume.EmptyDir != nil && volume.EmptyDir.Medium == v1.StorageMediumDefault) ||
```

Comment:
Can we assume that `volume.EmptyDir.Medium != v1.StorageMediumMemory` is equivalent to `volume.EmptyDir.Medium == v1.StorageMediumDefault`?

Reply:
No, we can't.  `volume.EmptyDir.Medium` is `v1.StorageMediumDefault`, `v1.StorageMediumMemory` or `v1.StorageMediumHugePages`.  `v1.StorageMediumHugePages` was added recently and `volume.EmptyDir.Medium` became ternary from binary. But `!=` in `IsLocalEphemeralVolume` and `==` in `isVolumeEphemeral` were not changed then. I think only `v1.StorageMediumDefault` is ephemeral, so I think the original code has a bug.

Reply:
According to [this](https://github.com/kubernetes/kubernetes/pull/55447#issue-151801859), HugePages is not a local ephemeral volume.

Reply:
why Secret volume is not an ephemeral volume?

Reply:
@cofyc Please refer this.
https://github.com/kubernetes/kubernetes/pull/82401#issuecomment-529743428

Reply:
Thanks!